project_test/analysis_file.py

```diff
@@ -0,0 +1,207 @@
+        """
+        For each TracedData object in an iterable whose 'withdrawn_key' is Codes.True, sets every other key to
+        Codes.STOP. If there is no withdrawn_key or the value is not Codes.True, that TracedData object is not modified.
+        
+        Arguments:
+            user {str} -- Identifier of the user running this program, for TracedData Metadata.
+            data {iterable of TracedData} -- TracedData objects to set to stopped if consent has been withdrawn.
+        
+        Keyword Arguments:
+            withdrawn_key {str} -- Key in each TracedData object which indicates whether consent has been withdrawn. (default: {"consent_withdrawn"})
+        """
+
+        for td in data:
+            if td.get(withdrawn_key) == Codes.TRUE:
+                stop_dict = {key: Codes.STOP for key in td.keys() if key != withdrawn_key}
+                td.append_data(stop_dict, Metadata(user, Metadata.get_call_location(), time.time()))       
+class AnalysisFile(object):
+    @staticmethod
+    def generate(user, data, csv_by_message_output_path, csv_by_individual_output_path):
+        # Serializer is currently overflowing
+        # TODO: Investigate/addres the cause of this.
+        sys.setrecursionlimit(10000)
+
+        consent_withdrawn_key = "consent_withdrawn"
+        for td in data:
+            td.append_data({consent_withdrawn_key: Codes.FALSE},
+                            Metadata(user, Metadata.get_call_location(), time.time()))
+
+        # Set the list of raw/coded keys
+        survey_keys = []
+        for plan in PipelineConfiguration.SURVEY_CODING_PLANS:
+            if plan.analysis_file_key not in survey_keys:
+                survey_keys.append(plan.analysis_file_key)
+            if plan.raw_field not in survey_keys:
+                survey_keys.append(plan.raw_field)
+        
+        # Convert survey codes to their string values
+        for td in data:
+            td.append_data(
+                {plan.analysis_file_key: plan.code_scheme.get_code_with_id(td[plan.coded_field]["CodeID"]).string_value
+                for plan in PipelineConfiguration.SURVEY_CODING_PLANS},
+                Metadata(user, Metadata.get_call_location(), time.time())
+            )
+
+        
+        # Convert Test Radio Questions binary codes to their string values
+        for td in data:
+            td.append_data(
+                {plan.binary_analysis_file_key:
+                plan.binary_code_scheme.get_code_with_id(td[plan.binary_coded_field]["CodeID"]).string_value
+                for plan in PipelineConfiguration.TEST_SHOWS_CODING_PLANS if plan.binary_code_scheme is not None},
+                Metadata(user, Metadata.get_call_location(), time.time())
+            )
+        
+        # Translate the Test Show Question to matrix values
+        matrix_keys = []
+
+        for plan in PipelineConfiguration.TEST_SHOWS_CODING_PLANS:
+            show_matrix_keys = set()
```

Comment:
(1/2) I recently changed this to a list in ADSS. This means you get the codes in scheme order rather than alphabetical order...